managed/services/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,383 @@
+	const (
+		tmpDistributionFile = "/tmp/distribution"
+		ami                 = "ami"
+	)
+	err := os.WriteFile(tmpDistributionFile, []byte(ami), fs.ModePerm)
+	if err != nil {
+		assert.Fail(t, "cannot write to file: ", err)
+		return nil
+	}
+	dus := newDistributionUtilServiceImpl(tmpDistributionFile, osInfoFilePath, l)
+	return dus
+}
+
+func initMockTelemetrySender(t *testing.T, expetedReport *reporter.ReportRequest, timesCall int) func() sender {
+	return func() sender {
+		var mockTelemetrySender mockSender
+		mockTelemetrySender.Test(t)
+		mockTelemetrySender.On("SendTelemetry",
+			mock.AnythingOfType(reflect.TypeOf(context.TODO()).Name()),
+			mock.MatchedBy(func(report *reporter.ReportRequest) bool {
+				return matchExpectedReport(report, expetedReport)
+			}),
+		).
+			Return(nil).
+			Times(timesCall)
+		t.Cleanup(func() {
+			mockTelemetrySender.AssertExpectations(t)
+		})
+
+		return &mockTelemetrySender
+	}
+}
+
+func matchExpectedReport(report *reporter.ReportRequest, expectedReport *reporter.ReportRequest) bool {
+	return len(report.Metrics) == 1 &&
+		expectedReport.Metrics[0].DistributionMethod.String() == "AMI"
+}
+
+func getTestConfig(sendOnStart bool, testSourceName string, reportingInterval time.Duration) ServiceConfig {
+	return ServiceConfig{
+		l:            nil,
+		Enabled:      true,
+		LoadDefaults: false,
+		telemetry: []Config{
+			{
+				ID:      "1",
+				Source:  testSourceName,
+				Query:   "pg_static{service_type=\"postgresql\"}",
+				Summary: "Monitored PostgreSQL services version",
+				Data: []ConfigData{
+					{
+						MetricName: "postgresql_version",
+						Label:      "short_version",
+					},
+				},
+			},
+		},
+		SaasHostname: "",
+		DataSources: struct {
+			VM          *DataSourceVictoriaMetrics `yaml:"VM"`
+			QanDBSelect *DSConfigQAN               `yaml:"QANDB_SELECT"`
+			PmmDBSelect *DSConfigPMMDB             `yaml:"PMMDB_SELECT"`
+		}{},
+		Reporting: ReportingConfig{
+			SendOnStart:     sendOnStart,
+			IntervalEnv:     "",
+			Interval:        reportingInterval,
+			RetryBackoffEnv: "",
+			RetryBackoff:    0,
+			SendTimeout:     0,
+			RetryCount:      3,
+		},
+	}
+}
+
+func initMockDB(t *testing.T, now time.Time, callTimes int) func() *reform.DB {
+	return func() *reform.DB {
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+
+		l := log.New(os.Stderr, "SQL: ", log.Flags())
+
+		var s models.Settings
+		b, err := json.Marshal(s)
+		assert.NoError(t, err)
+
+		for i := 0; i < callTimes; i++ {
+			initGetSettingsSQLMock(mock, b, now)
+		}
+
+		// we make sure that all expectations were met
+		t.Cleanup(func() {
+			if err := mock.ExpectationsWereMet(); err != nil {
+				t.Errorf("there were unfulfilled expectations for DB mock: %s", err)
+			}
+			defer db.Close() //nolint:errcheck
+		})
+		return reform.NewDB(db, postgresql.Dialect, reform.NewPrintfLogger(l.Printf))
+	}
+}
+
+func initGetSettingsSQLMock(mock sqlmock.Sqlmock, b []byte, now time.Time) {
```

Comment:
You can just write required values to real DB and read them. Like we do it here: https://github.com/percona/pmm/blob/1fe6469ab403800d2bd339b9ca3b23ca405f0bc2/managed/models/settings_helpers_test.go#L32

Reply:
Got it

Updated tests